src/components/sharedFields/HelmFields.tsx

```diff
@@ -25,112 +20,98 @@ const WithFlex = ({ flexItems }: { flexItems: React.ReactNode[] }) => {
     </Flex>
   );
 };
-
 const HelmFields = ({
   fieldNamePrefix,
-  horizontal = false,
+  reposListResult,
+  horizontal,
 }: {
+  reposListResult: HelmChartRepositoryListResult;
   fieldNamePrefix: string;
-  horizontal?: boolean;
+  horizontal: boolean;
 }) => {
-  const { t } = useTranslation();
-  const repoFieldName = `${fieldNamePrefix}.repo`;
   const chartFieldName = `${fieldNamePrefix}.chart`;
   const versionFieldName = `${fieldNamePrefix}.version`;
-  const [{ value: repo }, { error: repoError }] = useField<RepoOptionObject>(repoFieldName);
-  const [{ value: chart }, , { setValue: setChart }] = useField<string>(chartFieldName);
-  const [{ value: version }, , { setValue: setVersion }] = useField<string>(versionFieldName);
-  const [argoCDSecrets, argoCDSecretsLoaded, argoCDSecretsError] = useArgoCDSecrets();
-
-  const [helmRepository, helmRepositoryLoaded, helmRepositoryError] = useHelmChartRepository(
-    repo.toString(),
-  );
-
-  // t('Failed to load repositories')
-  useAddAlertOnError(argoCDSecretsError, 'Failed to load repositories');
-  // t('Failed to load charts')
-  useAddAlertOnError(helmRepositoryError, 'Failed to load charts');
-
-  const error = argoCDSecretsError || helmRepositoryError;
-
-  const repoOptions: SelectInputOption[] = React.useMemo(
-    () =>
-      argoCDSecrets.map((s) => ({
-        value: getRepoOptionObject(s),
-        disabled: false,
-      })),
-    [argoCDSecrets],
-  );
+  const { values, setFieldValue } = useFormikContext();
+  const { t } = useTranslation();
+  const [chartToVersions, setChartToVersions] = React.useState<Record<string, string[]>>();
+  const chart = get(values, chartFieldName);
+  const [, loaded] = reposListResult;
+  const clearChartFields = () => {
+    setFieldValue(chartFieldName, '');
+    setFieldValue(versionFieldName, '');
+  };
 
-  const chartsMap = React.useMemo<HelmRepositoryChartsMap>(() => {
-    if (!helmRepository?.index || !repo) {
-      return {};
+  const setRepo = (repo: HelmRepository): Record<string, string[]> => {
+    let chartToVersions: Record<string, string[]> = undefined;
+    if (!repo.index) {
+      clearChartFields();
+    } else {
+      const map: Record<string, string[]> = {};
+      for (const [key, charts] of Object.entries(repo.index.entries)) {
+        map[key] = charts
+          .map((chart) => chart.version)
+          .sort((v1, v2) => v2.localeCompare(v1, undefined, { numeric: true }));
+      }
+      chartToVersions = map;
     }
-    return getRepoChartsMap(helmRepository.index);
-  }, [helmRepository?.index, repo]);
+    setChartToVersions(chartToVersions);
```

Comment:
Does this need to be kept in state? Can't we always calculate it from the form value? I assume that we don't allow multiple repositories with the same URL. Alternatively we can have a repository name as a form value instead of the URL and transform it when submitting the wizard.